### ISSUE TITLE
Expose a stable public accessor for fitted parameter vectors in penaltyblog goal-scoring models

### ISSUE THREAD

**Issue by ssoulless** [martineastwood/penaltyblog#14]
#### 1 · Background & motivation

When building post-fit tooling we often need direct access to a model’s fitted parameters:

```python
model = pb.models.PoissonGoalsModel(...)
model.fit()

# Today: rely on the private attribute
p = model._params        # <- private, can change without notice
```

Accessing `_params` breaks encapsulation and triggers static-analysis warnings (`_params` is considered implementation detail). It also makes downstream libraries fragile if the internal layout changes.

#### 2 · Typical downstream use cases

* Re-computing λ<sub>home</sub>/λ<sub>away</sub> after applying multiplicative factors.
* Running Monte-Carlo simulations or custom likelihoods outside the built-in Cython helpers.
* Vectorised batch evaluation for Optuna or other hyper-parameter searches.

All of those require the full parameter vector (or at least a documented subset: attack, defence, home-field advantage, plus model-specific extras such as `rho`, `dispersion`, `shape`, `kappa`, etc.).

#### 3 · Proposal

Add a lightweight public accessor that returns a *read-only* copy of the parameters:

```python
class PoissonGoalsModel(...):
    ...
    @property
    def params(self) -> np.ndarray:
        """Return the fitted parameter vector in the same order used internally."""
        return self._params.copy()
```

* **Consistent name** across all goal-scoring models (`Poisson`, `DixonColes`, `NegativeBinomial`, `ZeroInflatedPoisson`, `Bivariate`, `WeibullCopula`, …).
* Document the ordering of elements (or return a named tuple / dataclass if you prefer stronger typing).
* Keep `_params` untouched for backward compatibility; `params` simply exposes a copy.

#### 4 · Benefits

* Downstream code (including ours) can drop the private access and silence linters.
* Future refactors of the internal storage remain possible as long as the public contract is kept.
* Encourages ecosystem growth: external libraries can build adapters, visualisers and batch evaluators on top of `penaltyblog` without brittle hacks.

#### 5 · Environment

* `penaltyblog` == 1.4.1
* Python 3.13
* NumPy 2.2.4

---

Happy to open a PR with the minimal implementation + docs if the approach is acceptable, want it to discuss first before coding it. Thanks
